src/main/java/net/floodlightcontroller/sos/SOSDevice.java

```diff
@@ -18,6 +19,12 @@ public SOSDevice(SOSDeviceType t, IPv4Address ip) {
 		mac_addr = MacAddress.NONE;
 		type = t;
 	}
+	public SOSDevice(SOSDeviceType t, IPv4Address ip, IPv4Address restIPAddr) {
+		ip_addr = ip;
+		rest_ip_addr = restIPAddr;
+		mac_addr = MacAddress.NONE;
+		type = t;
+	}
 	public SOSDevice(SOSDeviceType t, IPv4Address ip, MacAddress mac) {
```

Comment:
In constructors where the rest ip isn't set, you should set it to some default value, just in case someone tries to retrieve it

Reply:
Good Point. Fixed .